tutorials/persistance-comparison-series/PlayerPrefs/player_prefs.md

```diff
@@ -0,0 +1,436 @@
+[SerializeField] private int hitCountControl = 0;
+```
+
+All those will eventually be saved into the same `PlayerPrefs` field which means we only need one key (3):
+
+```cs
+// 3
+private readonly string hitCountKey = "HitCountKeyJson";
+```
+
+As before, the `keyPressed` will indicate which modifier was used:
+
+```cs
+// 4
+private KeyCode keyPressed = default;
+```
+
+In `Start()` we then need to read the JSON. As before, we check if the `PlayerPrefs` key exists (5) and then read the data, this time using `GetString()` (as opposed to `GetInt()` before).
+
+Transforming this JSON into the actual object is then done using `JsonUtility.FromJson()` (6) which takes the string as an argument. It's a generic function and we need to provide the information about which object this JSON is supposed to be representing, in this case `HitCount`.
+
+If the JSON could be read and transformed successfully we can set the hit count fields (7) to their three values.
+
+```cs
+private void Start()
+{
+    // 5
+    // Check if the key exists. If not, we never saved to it.
+    if (PlayerPrefs.HasKey(HitCountKey))
+    {
+        // 6
+        string jsonString = PlayerPrefs.GetString(HitCountKey);
+        HitCount hitCount = JsonUtility.FromJson<HitCount>(jsonString);
+
+        // 7
+        if (hitCount != null)
+        {
+            hitCountUnmodified = hitCount.Unmodified;
+            hitCountShift = hitCount.Shift;
+            hitCountControl = hitCount.Control;
+        }
+    }
+}
+```
+
+The detection for the key that was pressed is identical to the extended example since it does not involve loading or saving any data but is just a check for the key during `Update()`:
+
+```cs
+private void Update() // 8
+{
+    // Check if a key was pressed.
+    if (Input.GetKey(KeyCode.LeftShift)) // 9
+    {
+        // Set the LeftShift key.
+        keyPressed = KeyCode.LeftShift; // 10
+    }
+    else if (Input.GetKey(KeyCode.LeftControl)) // 9
+    {
+        // Set the LeftControl key.
+        keyPressed = KeyCode.LeftControl; // 10
+    }
+    else // 11
+    {
+        // In any other case reset to default and consider it unmodified.
+        keyPressed = default; // 12
+    }
+}
+```
+
+In a very similar fashion `OnMouseDown()` needs to save the data whenever it's changed.
+
+```cs
+private void OnMouseDown()
+{
+    if (keyPressed == KeyCode.LeftShift) // 13
+    {
+        // Increment the hit count and set it to PlayerPrefs.
+        hitCountShift++; // 14
+    }
+    else if (keyPressed == KeyCode.LeftControl) // 13
+    {
+        // Increment the hit count and set it to PlayerPrefs.
+        hitCountControl++; // 14
+    }
+    else // 15
+    {
+        // Increment the hit count and set it to PlayerPrefs.
+        hitCountUnmodified++; // 16
+    }
+
+    // 17
+    HitCount hitCount = new();
+    hitCount.Unmodified = hitCountUnmodified;
+    hitCount.Shift = hitCountShift;
+    hitCount.Control = hitCountControl;
```

Comment:
This can be simplified a little by using the property initializer syntax:

```suggestion
    var updatedCount = new HitCount
    { 
        Unmodified = hitCountUnmodified,
        Shift = hitCountShift,
        Control = hitCountControl,
    };
```